packages/graph/hash_graph/lib/graph/src/store/postgres/query/expression/join_clause.rs

```diff
@@ -26,15 +19,12 @@ impl Transpile for JoinExpression<'_> {
     fn transpile(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
         // TODO: https://app.asana.com/0/1202805690238892/1203324626226299/f
         fmt.write_str("INNER JOIN ")?;
-        if self.join.table.alias.is_some() {
-            let unaliased_table = Table {
-                name: self.join.table.name,
-                alias: None,
-            };
-            unaliased_table.transpile(fmt)?;
+
+        if self.join.alias.is_some() {
+            self.join.column.table().transpile(fmt)?;
             fmt.write_str(" AS ")?;
         }
-        self.join.table.transpile(fmt)?;
+        self.join.table().transpile(fmt)?;
```

Comment:
Couple quick question for the aliasing here:
In the case that we don't have an alias, would the table name be used for the join case?

We are not using writing the alias directly here, is that intentional? Is it implicit in the transpile?

Reply:
Yes, that is intentional, it will generate something like `table_name AS table_name_<ALIAS>`. The `table_name_<ALIAS>` is implicit in the `table()`.